homeassistant/components/goalzero/config_flow.py

```diff
@@ -0,0 +1,75 @@
+"""Config flow for Goal Zero Yeti integration."""
+import logging
+
+from goalzero import Yeti, exceptions
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DEFAULT_NAME, DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema({"host": str, "name": str})
+
+
+class GoalZeroFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Goal Zero Yeti."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initiated by the user."""
+        errors = {}
+
+        if user_input is not None:
+            host = user_input[CONF_HOST]
+            name = user_input[CONF_NAME]
+
+            if await self._async_endpoint_existed(host):
+                return self.async_abort(reason="already_configured")
+
+            try:
+                await self._async_try_connect(host)
+                return self.async_create_entry(
```

Comment:
Please only wrap the line that can raise in the try... except block. Use an `else:` block if needed.